From b09ccf1009aec82ca7cbb3d86e7a134fbab20f76 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 21 Nov 2013 20:14:53 -0500
Subject: [PATCH] Do not verify session for bungeecord connections

This allows you to keep online-mode=true for bungee.
---
 src/main/java/net/minecraft/server/ThreadLoginVerifier.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main/java/net/minecraft/server/ThreadLoginVerifier.java b/src/main/java/net/minecraft/server/ThreadLoginVerifier.java
index 70f9fea..12c1220 100644
--- a/src/main/java/net/minecraft/server/ThreadLoginVerifier.java
+++ b/src/main/java/net/minecraft/server/ThreadLoginVerifier.java
@@ -28,6 +28,7 @@ class ThreadLoginVerifier extends Thread {
     }
 
     private boolean auth() throws java.io.IOException {
+        if (org.spigotmc.SpigotConfig.bungee && org.spigotmc.SpigotConfig.bungeeAddresses.contains(this.pendingConnection.getSocket().getInetAddress().getHostAddress())) { return true; } // Spigot
         String s = (new BigInteger(MinecraftEncryption.a(PendingConnection.a(this.pendingConnection), PendingConnection.b(this.pendingConnection).H().getPublic(), PendingConnection.c(this.pendingConnection)))).toString(16);
         URL url = new URL("http://session.minecraft.net/game/checkserver.jsp?user=" + URLEncoder.encode(PendingConnection.d(this.pendingConnection), "UTF-8") + "&serverId=" + URLEncoder.encode(s, "UTF-8"));
         BufferedReader bufferedreader = new BufferedReader(new InputStreamReader(url.openConnection(PendingConnection.b(this.pendingConnection).ap()).getInputStream()));
-- 
1.8.4.4

